x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/CategoryDefinition.java

```diff
@@ -90,7 +90,7 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeString(terms);
         out.writeString(regex);
         out.writeLong(maxMatchingLength);
-        out.writeStringList(new ArrayList<>(examples));
```

Comment:
It's a bit odd that this was previously wrapped in a new `ArrayList` instance but I don't see a reason for that so your change is fine.

Reply:
It is because `examples` is only a `TreeSet`, which means that we can not use the existing `writeStringList` to serialize this connection. Instead, it is converted to a list by constructing a new one from `examples`. The benefit of this change is that by liberalizing `writeStringList` to write an arbitrary collection and not only a list, we no longer need to do that conversion.

Reply:
Right! Makes perfect sense now.